src/sql/workbench/services/objectExplorer/browser/asyncServerTreeDataSource.ts

```diff
@@ -51,11 +51,7 @@ export class AsyncServerTreeDataSource implements IAsyncDataSource<ConnectionPro
 			} else if (element instanceof ConnectionProfileGroup) {
 				return element.getChildren();
 			} else if (element instanceof TreeNode) {
-				if (element.children) {
-					return element.children;
```

Comment:
@Charles-Gagnon , can provide more context about why it was added in the first place. However, in case of updating children we should be fetching the child nodes from oe service again, but this prevented that.